graphene_sqlalchemy/enums.py

```diff
@@ -0,0 +1,204 @@
+    return enum
+
+
+def _default_sort_enum_symbol_name(column_name, sort_asc=True):
+    return to_enum_value_name(column_name) + ("_ASC" if sort_asc else "_DESC")
+
+
+def sort_enum_for_object_type(
+    obj_type, name=None, only_fields=None, only_indexed=None, get_symbol_name=None
+):
+    """Return Graphene Enum for sorting the given SQLAlchemyObjectType.
+
+    Parameters
+    - obj_type : SQLAlchemyObjectType
+        The object type for which the sort Enum shall be generated.
+    - name : str, optional, default None
+        Name to use for the sort Enum.
+        If not provided, it will be set to the object type name + 'SortEnum'
+    - only_fields : sequence, optional, default None
+        If this is set, only fields from this sequence will be considered.
+    - only_indexed : bool, optional, default False
+        If this is set, only indexed columns will be considered.
+    - get_symbol_name : function, optional, default None
+        Function which takes the column name and a boolean indicating
+        if the sort direction is ascending, and returns the symbol name
+        for the current column and sort direction. If no such function
+        is passed, a default function will be used that creates the symbols
+        'foo_asc' and 'foo_desc' for a column with the name 'foo'.
+
+    Returns
+    - Enum
+        The Graphene Enum type
+    """
+    name = name or obj_type._meta.name + "SortEnum"
+    registry = obj_type._meta.registry or get_global_registry()
+    enum = registry.get_sort_enum_for_object_type(obj_type)
+    custom_options = dict(
+        only_fields=only_fields,
+        only_indexed=only_indexed,
+        get_symbol_name=get_symbol_name,
+    )
+    if enum:
+        if name != enum.__name__ or custom_options != enum.custom_options:
+            raise ValueError(
+                "Sort enum for {} has already been customized".format(obj_type)
+            )
+    else:
+        members = []
+        default = []
+        fields = obj_type._meta.fields
+        get_name = get_symbol_name or _default_sort_enum_symbol_name
+        for field_name in fields:
+            if only_fields and field_name not in only_fields:
+                continue
+            orm_field = registry.get_orm_field_for_graphene_field(obj_type, field_name)
+            if not isinstance(orm_field, Column):
+                continue
+            if only_indexed and not (orm_field.primary_key or orm_field.index):
+                continue
+            asc_name = get_name(orm_field.name, True)
+            asc_value = EnumValue(asc_name, orm_field.asc())
+            desc_name = get_name(orm_field.name, False)
+            desc_value = EnumValue(desc_name, orm_field.desc())
+            if orm_field.primary_key:
+                default.append(asc_value)
+            members.extend(((asc_name, asc_value), (desc_name, desc_value)))
+        enum = Enum(name, members)
+        enum.default = default  # store default as attribute
```

Comment:
Does it make sense to subclass Enum here?

Reply:
You mean something like SortEnumWithDefaultOrder?